code/alpaka_interface/AllocatorConfig.h

```diff
@@ -0,0 +1,31 @@
+#ifndef HeterogeneousCore_AlpakaInterface_interface_AllocatorConfig_h
+#define HeterogeneousCore_AlpakaInterface_interface_AllocatorConfig_h
+
+#include <limits>
+
+namespace cms::alpakatools {
```

Comment:
This namespace won't work - it conflicts with the CMSSW one... Can we switch to `lst::`?

Reply:
> Can we switch to `lst::`?

I tried this but it didn't solve the initial error. I am sending the initial error [0]. Just to make it clear, the only thing I get out of LST for CMSSW is the `libsdl.so`. The include paths seem to be ok.

[0]:
```
In file included from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getDeviceCachingAllocator.h:9,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:6,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Constants.h:5,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Hit.h:4,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Event.h:4,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/LST.h:11,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/RecoTracker/LST/plugins/alpaka/LSTProducer.cc:22:
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachingAllocator.h:87:59: error: 'is_device_v' is not a member of 'cms::alpakatools'
   87 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                           ^~~~~~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachingAllocator.h:87:75: error: template argument 1 is invalid
   87 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                           ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachingAllocator.h:87:77: error: expected '>' before 'and' token
   87 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                             ^~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachingAllocator.h:87:118: error: expected unqualified-id before '>' token
   87 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                                                                      ^
In file included from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:6,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Constants.h:5,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Hit.h:4,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Event.h:4,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/LST.h:11,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/RecoTracker/LST/plugins/alpaka/LSTProducer.cc:22:
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getDeviceCachingAllocator.h:19:61: error: 'is_device_v' is not a member of 'cms::alpakatools'
   19 |               typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                             ^~~~~~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getDeviceCachingAllocator.h:19:77: error: template argument 1 is invalid
   19 |               typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                             ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getDeviceCachingAllocator.h:19:79: error: expected '>' before 'and' token
   19 |               typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                               ^~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getDeviceCachingAllocator.h:19:120: error: expected unqualified-id before '>' token
   19 |               typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                                                                        ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getDeviceCachingAllocator.h:75:59: error: 'is_device_v' is not a member of 'cms::alpakatools'
   75 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                           ^~~~~~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getDeviceCachingAllocator.h:75:75: error: template argument 1 is invalid
   75 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                           ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getDeviceCachingAllocator.h:75:77: error: expected '>' before 'and' token
   75 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                             ^~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getDeviceCachingAllocator.h:75:118: error: expected unqualified-id before '>' token
   75 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                                                                      ^
In file included from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:7,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Constants.h:5,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Hit.h:4,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Event.h:4,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/LST.h:11,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/RecoTracker/LST/plugins/alpaka/LSTProducer.cc:22:
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getHostCachingAllocator.h:13:76: error: 'is_queue_v' is not a member of 'cms::alpakatools'
   13 |   template <typename TQueue, typename = std::enable_if_t<cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                            ^~~~~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getHostCachingAllocator.h:13:87: error: template argument 1 is invalid
   13 |   template <typename TQueue, typename = std::enable_if_t<cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                                       ^~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/getHostCachingAllocator.h:13:95: error: expected unqualified-id before '>' token
   13 |   template <typename TQueue, typename = std::enable_if_t<cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                                               ^
In file included from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Constants.h:5,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Hit.h:4,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Event.h:4,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/LST.h:11,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/RecoTracker/LST/plugins/alpaka/LSTProducer.cc:22:
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:21:61: error: 'is_device_v' is not a member of 'cms::alpakatools'
   21 |               typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                             ^~~~~~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:21:77: error: template argument 1 is invalid
   21 |               typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                             ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:21:79: error: expected '>' before 'and' token
   21 |               typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                               ^~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:21:120: error: expected unqualified-id before '>' token
   21 |               typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                                                                        ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:28:12: error: 'CachedBufAlloc' is not a class template
   28 |     struct CachedBufAlloc<TElem, TDim, TIdx, alpaka::DevCpu, TQueue, void> {
      |            ^~~~~~~~~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:41:95: error: wrong number of template arguments (6, should be 4)
   41 |     struct CachedBufAlloc<TElem, TDim, TIdx, alpaka::DevCpu, alpaka::QueueCudaRtBlocking, void> {
      |                                                                                               ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:28:74: note: provided for 'template<class TElem, class TDim, class TIdx, class TQueue> struct cms::alpakatools::traits::CachedBufAlloc'
   28 |     struct CachedBufAlloc<TElem, TDim, TIdx, alpaka::DevCpu, TQueue, void> {
      |                                                                          ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:64:98: error: wrong number of template arguments (6, should be 4)
   64 |     struct CachedBufAlloc<TElem, TDim, TIdx, alpaka::DevCpu, alpaka::QueueCudaRtNonBlocking, void> {
      |                                                                                                  ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:28:74: note: provided for 'template<class TElem, class TDim, class TIdx, class TQueue> struct cms::alpakatools::traits::CachedBufAlloc'
   28 |     struct CachedBufAlloc<TElem, TDim, TIdx, alpaka::DevCpu, TQueue, void> {
      |                                                                          ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:87:77: error: wrong number of template arguments (6, should be 4)
   87 |     struct CachedBufAlloc<TElem, TDim, TIdx, alpaka::DevCudaRt, TQueue, void> {
      |                                                                             ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:28:74: note: provided for 'template<class TElem, class TDim, class TIdx, class TQueue> struct cms::alpakatools::traits::CachedBufAlloc'
   28 |     struct CachedBufAlloc<TElem, TDim, TIdx, alpaka::DevCpu, TQueue, void> {
      |                                                                          ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:200:59: error: 'is_device_v' is not a member of 'cms::alpakatools'
  200 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                           ^~~~~~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:200:75: error: template argument 1 is invalid
  200 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                           ^
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:200:77: error: expected '>' before 'and' token
  200 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                             ^~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/../code/alpaka_interface/CachedBufAlloc.h:200:118: error: expected unqualified-id before '>' token
  200 |             typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev> and cms::alpakatools::is_queue_v<TQueue>>>
      |                                                                                                                      ^
In file included from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Event.h:6,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/LST.h:11,
                 from /home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/RecoTracker/LST/plugins/alpaka/LSTProducer.cc:22:
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Segment.h: In function 'float SDL::isTighterTiltedModules_seg(SDL::modules&, unsigned int)':
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Segment.h:187:33: warning: suggest parentheses around comparison in operand of '&' [-Wparentheses]
  187 |                         | (side == NegZ & (((layer == 2) & (rod > 5)) | ((layer == 1) & (rod > 9))))
      |                            ~~~~~^~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Segment.h:188:33: warning: suggest parentheses around comparison in operand of '&' [-Wparentheses]
  188 |                         | (side == PosZ & (((layer == 2) & (rod < 8)) | ((layer == 1) & (rod < 4)))) );
      |                            ~~~~~^~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Segment.h:186:23: warning: suggest parentheses around comparison in operand of '&' [-Wparentheses]
  186 |         return subdet == Barrel & ( ((side != Center) & (layer== 3))
      |                ~~~~~~~^~~~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Segment.h: In function 'float SDL::isTighterTiltedModules_seg(short int, short int, short int, short int)':
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Segment.h:197:33: warning: suggest parentheses around comparison in operand of '&' [-Wparentheses]
  197 |                         | (side == NegZ & (((layer == 2) & (rod > 5)) | ((layer == 1) & (rod > 9))))
      |                            ~~~~~^~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Segment.h:198:33: warning: suggest parentheses around comparison in operand of '&' [-Wparentheses]
  198 |                         | (side == PosZ & (((layer == 2) & (rod < 8)) | ((layer == 1) & (rod < 4)))) );
      |                            ~~~~~^~~~~~~
/home/users/evourlio/LSTinCMSSW/cgpu-1/CMSSW_13_0_0_pre4/src/../../../TrackLooper/SDL/Segment.h:196:23: warning: suggest parentheses around comparison in operand of '&' [-Wparentheses]
  196 |         return subdet == Barrel & ( ((side != Center) & (layer== 3))
      |                ~~~~~~~^~~~~~~~~
gmake: *** [config/SCRAM/GMake/Makefile.rules:1740: tmp/el8_amd64_gcc11/src/RecoTracker/LST/plugins/RecoTrackerLSTPluginsPortableCudaAsync/alpaka/LSTProducer.cc.o] Error 1
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2

```

Reply:
From the error, it doesn’t seem like you changed all of the cms::alpakatools in the directory to lst::, I’ll push a fix